KSR_002/KSR_002.md

```diff
@@ -0,0 +1,282 @@
+
+- clock_gettime;
+- gettimeofday;
+- getcpu;
+- time;
+- clock_getres;
+
+**Conclusion: vdso** functions are not relevant for the safety app as they are not used at all by the safety app
+
+**Conclusion: vvar (vdso variable) **are not relevant for the safety app as VDSO functions are not used by the safety app
+
+<mark > **TODO: to formally confirm this, eventually using the calltree tool (or cscope tool)** </mark>
+
+The [modified safety app](https://github.com/paolonig/wg_automotive_fork/blob/test_branch/safety-app.c) code has been annotated to mark the code that is optional (i.e. can be removed without affecting the functionalities of the safety app itself) or that is not active during	the safe operation. After the annotations the following glibc calls are left to be analyzed:
+
+- [mkfifo()](https://github.com/paolonig/wg_automotive_fork/blob/test_branch/safety-app.c#L70): looking at the glibc source code the call-tree is quite simple:  [__mknod](https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/posix/mkfifo.c;h=5f9452ce497a6bbd6bc42ea23c4c4af7118c2e09;hb=HEAD#l27)()->__mknod() ->mknodat(). Hence here there is not involvement at all of vdso and vvar
+- [open()](https://github.com/paolonig/wg_automotive_fork/blob/test_branch/safety-app.c#L71): glibc implements a wrapper for the respective kernel syscall
+- [ioctl()](https://github.com/paolonig/wg_automotive_fork/blob/test_branch/safety-app.c#L79): glibc implements a wrapper for the respective kernel syscall
+- [usleep()](https://github.com/paolonig/wg_automotive_fork/blob/test_branch/safety-app.c#L109): 
+
+<table>
+  <tr>
+   <td>
+usleep
+   </td>
+   <td>https://elixir.bootlin.com/glibc/glibc-2.35/source/sysdeps/posix/usleep.c#L23
+   </td>
+  </tr>
+  <tr>
+   <td>__nanosleep
+   </td>
+   <td>https://elixir.bootlin.com/glibc/glibc-2.35/source/posix/nanosleep.c#L24
+   </td>
+  </tr>
+  <tr>
+   <td>__clock_nanosleep
+   </td>
+   <td>https://elixir.bootlin.com/glibc/glibc-2.35/source/sysdeps/unix/sysv/linux/clock_nanosleep.c#L78
+   </td>
+  </tr>
+  <tr>
+   <td>__clock_nanosleep_time64
+   </td>
+   <td>https://elixir.bootlin.com/glibc/glibc-2.35/source/sysdeps/unix/sysv/linux/clock_nanosleep.c#L30
+   </td>
+  </tr>
+</table>
+
+
+
+`__clock_nanosleep_time64()` implements a wrapper around `clock_nanosleep_time64` or `clock_nanosleep` syscalls according with the underlying architecture.
+
+
+
+- [fflush()](https://github.com/paolonig/wg_automotive_fork/blob/test_branch/safety-app.c#L111): TBD
+
+<mark > **TODO:** </mark> there are lots of objects mapped in the safety app address space [here](https://docs.google.com/document/d/1PH9cVmuWjbRLjbRvz8iWBuJBhgpkUPclnj1IpwRxoBs/edit). We need to	understand which of these objects (and related areas) are safety related and which are not. Using [prelink](https://linux.die.net/man/8/prelink)() we can possibly retrieve the symbols required by the program I want to run (so we can classify accordingly the rows of the proc/pid/maps)
```

Comment:
The proposed method is obsolete. From the email here "https://lists.elisa.tech/g/safety-architecture/message/667" we can use nm -D "executable_name" in order to retrieve the symbols invoked from the linked objects